doc/sources/algorithms.rst

```diff
@@ -20,177 +20,331 @@
+       - ``sample_weight`` != `None`
+       - ``penalty`` != `'l2'`
+     - Only dense data is supported
+
+Regression
+**********
+
 .. list-table::
-   :widths: 10 10 30 15
+   :widths: 10 30 20
    :header-rows: 1
    :align: left
 
-   * - Task
-     - Functionality
-     - Parameters support
-     - Data support
-   * - Classification
-     - SVC
-     - All parameters except ``kernel`` = 'sigmoid_poly', ``class_weight`` != None.
-     - Only binary dense data is supported.
-   * - Classification
-     - RandomForestClassifier
-     - All parameters except ``warm_start`` = True, ``cpp_alpha`` != 0, ``criterion`` != 'gini', ``oob_score`` = True.
-     - Multi-output, sparse data, out-of-bag score and sample_weight are not supported.
-   * - Classification
-     - KNeighborsClassifier
-     - All parameters except ``algorithm`` != 'brute', ``weights`` = 'callable', ``metric`` not in ['euclidean', 'manhattan', 'minkowski', 'chebyshev', 'cosine'].
-     - Only dense data is supported.
-   * - Classification
-     - LogisticRegression
-     - All parameters except ``solver`` != 'newton-cg', ``class_weight`` != None, ``sample_weight`` != None, ``penalty`` != 'l2'
-     - Only dense data is supported.
-   * - Regression
-     - RandomForestRegressor
-     - All parameters except ``warm_start`` = True, ``cpp_alpha`` != 0, ``criterion`` != 'mse', ``oob_score`` = True.
-     - Multi-output, sparse data, out-of-bag score and sample_weight are not supported.
-   * - Regression
-     - KNeighborsRegressor
-     - All parameters except ``algorithm`` != 'brute', ``weights`` = 'callable', ``metric`` != 'euclidean' or 'minkowski' with ``p`` != 2.
-     - Only dense data is supported.
-   * - Regression
-     - LinearRegression
-     - All parameters except ``normalize`` != False and ``sample_weight`` != None.
-     - Only dense data is supported, #observations should be >= #features.
-   * - Clustering
-     - KMeans
-     - All parameters except ``precompute_distances`` and ``sample_weight`` != None. Init = 'k-means++' fallbacks to CPU.
-     - Sparse data is not supported.
-   * - Clustering
-     - DBSCAN
-     - All parameters except ``metric`` != 'euclidean', ``algorithm`` != 'brute', ``algorithm`` != 'auto'.
-     - Only dense data is supported.
-   * - Dimensionality reduction
-     - PCA
-     - All parameters except ``svd_solver`` != 'full'.
-     - Sparse data is not supported.
-   * - Unsupervised
-     - NearestNeighbors
-     - All parameters except ``algorithm`` != 'brute', ``weights`` = 'callable', ``metric`` not in ['euclidean', 'manhattan', 'minkowski', 'chebyshev', 'cosine'].
-     - Only dense data is supported.
+   * - Algorithm
+     - Parameters
+     - Data formats
+   * - `RandomForestRegressor`
+     - All parameters are supported except:
+
+       - ``warm_start`` = `True`
+       - ``cpp_alpha`` != `0`
+       - ``criterion`` != `'mse'`
+       - ``oob_score`` = `True`
+       - ``sample_weight``
```

Comment:
I've moved the info about sample_weight from the data column to the parameters column. question: should it say `sample_weight != None` or  it's ok as it is?

Reply:
Ok, let's align in all algorithms and use **sample_weight != None**, new place (in parameter column) is correct too

Reply:
> Ok, let's align in all algorithms and use **sample_weight != None**, new place (in parameter column) is correct too

ok, I'll make these updates

Reply:
done